VitruvioHost/Plugins/Vitruvio/Source/Vitruvio/Private/VitruvioModule.cpp

```diff
@@ -379,45 +386,53 @@ Vitruvio::FTextureData VitruvioModule::DecodeTexture(UObject* Outer, const FStri
 	return Vitruvio::DecodeTexture(Outer, Key, Path, TextureMetadata, std::move(Buffer), BufferSize);
 }
 
-FBatchGenerateResult VitruvioModule::BatchGenerateAsync(TArray<FInitialShape> InitialShapes, URulePackage* RulePackage) const
+FBatchGenerateResult VitruvioModule::BatchGenerateAsync(TArray<FInitialShape> InitialShapes) const
 {
-	check(RulePackage);
-    
     const FBatchGenerateResult::FTokenPtr Token = MakeShared<FGenerateToken>();
     	
 	CHECK_PRT_INITIALIZED_ASYNC(FBatchGenerateResult, Token)
 
-	FBatchGenerateResult::FFutureType ResultFuture = Async(EAsyncExecution::Thread, [=, InitialShapes = std::move(InitialShapes)]() mutable {
-		FGenerateResultDescription Result = BatchGenerate(InitialShapes, RulePackage);
+	FBatchGenerateResult::FFutureType ResultFuture = Async(EAsyncExecution::Thread, [=, InitialShapes = MoveTemp(InitialShapes)]() mutable {
+		FGenerateResultDescription Result = BatchGenerate(MoveTemp(InitialShapes));
 		return FBatchGenerateResult::ResultType { Token, MoveTemp(Result) };
 	});
 
 	return FBatchGenerateResult { MoveTemp(ResultFuture), Token };
 }
 
-FGenerateResultDescription VitruvioModule::BatchGenerate(const TArray<FInitialShape>& InitialShapes, URulePackage* RulePackage) const
+FGenerateResultDescription VitruvioModule::BatchGenerate(TArray<FInitialShape> InitialShapes) const
 {
-    check(RulePackage);
-
     CHECK_PRT_INITIALIZED()
 
 	GenerateCallsCounter.Add(InitialShapes.Num());
 
-    const ResolveMapSPtr ResolveMap = LoadResolveMapAsync(RulePackage).Get();
+	TMap<URulePackage*, TArray<FInitialShape>> RulePackages;
```

Comment:
The name is a bit ambiguous for a map. Maybe we could name it more specific i.e. RulePackageToInitialShape, RuleToShape, ect..